src/sys/stat.rs

```diff
@@ -109,3 +111,62 @@ pub fn fstat(fd: RawFd) -> Result<FileStat> {
 
     Ok(dst)
 }
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod consts {
+    use libc::c_int;
+
+    bitflags!(
+        flags FstatatFlag: c_int {
```

Comment:
I saw that nix unify some flags with the same name prefix, but I did not understand the reasoning here. Because some flags are only accepted by some system calls. For instance `AT_EMPTY_PATH` is not accepted by `fchownat` and even for the same function different operating systems differ in flags they accept for a system call wrapper. Would this not lead to subtle bugs, when porting applications?